### DISCUSSION TITLE
Export format does not match Import format

### DISCUSSION THREAD

**Discussion by Hatric** [netbox-community/netbox#12388] [Q&A]
We are working on standing up Netbox as our source of truth for Inventory + IPAM.  We have thousands of devices from 10's of groups to start training on importing their content.

We were planning to leverage Export CSV to provide a Template to distribute to users who already have equipment stored in Excel format, and request that they fill in the Template for Importing into the new installation of Netbox.

While doing this, we've noticed that when we Export to CSV - the format does not match the Import Format.  It seems strange to have different formatting between the two.  Is there a recommendation or setting we're missing when exporting data?

We can go through editing the template prior to sharing, but just thought it was a bit strange to have to do this and wondered if there was a reason other than friendly naming.

**Comment by abhi1693** [comment]
You can refer to the FAQ for this: https://github.com/netbox-community/netbox/wiki/Frequently-Asked-Questions#why-cant-i-pass-exported-data-directly-into-the-import-form

**Comment by PenelopeFudd** [comment]
The easiest way to generate data that's acceptable to an import function is to export the data and modify it.

In our case, it would have helped us figure out that several fields (e.g. Tenant) were case-sensitive.

Being able to round-trip the data is also a quick-and-dirty way to do backups/restores of parts of the database when you're iterating on a desired layout: Export to file, edit file, delete records from netbox, import file, and repeat until satisfied.  It's akin to the Edit/Compile/Test cycle.

Oh, and when importing, I like the fact that the error message prints which record isn't acceptable.  It would be even better if it also highlighted the failing record(s) or had line numbers on the import box.

Thanks!

**Reply by Hatric** [reply]
+1 It would be even better if it also highlighted the failing record(s) or had line numbers on the import box.

Yea when you're working with 100+ rows, the numbers would be great.
